pkgs/os-specific/linux/nvidia-x11/default.nix

```
@@ -22,10 +22,12 @@ rec {
   # Policy: use the highest stable version as the default (on our master).
   stable = if stdenv.hostPlatform.system == "x86_64-linux"
     then generic {
-      version = "440.59";
-      sha256_64bit = "162gq6w44l8sgnn4qnl2rdlx8c008p04zv4c3i1ps20p21n1mjv1";
-      settingsSha256 = "0vxhmirqzyav5ljf0f04yk0az48ir5v0817dq9z9kyqfdvnby93g";
-      persistencedSha256 = "0npjh7nashasydp8q6bbcp21w8fc1dycgjy50ics775hjnvm61qn";
+      version = "440.64";
+      sha256_64bit = "0xbm1dh95kz8h4d62pql2wmvw2gbgc7iif2bkixbnqijl4dryg71";
+      settingsSha256 = "1vdir8a8cky4kmipqsbyjhjn0aqbwgzsxq73hafikrp5n4nbclfh";
+      persistencedSha256 = "0lcnila7xyl5r87c88sq2fn5k6ylmdf1bk2wcvm6aw5x6pmnrkgi";
+
+      patches = [ ./patches/linux-version.patch ./patches/kernel-5.6.patch ];
```

```suggestion
      patches = [
        (fetchpatch {
          url = "https://raw.githubusercontent.com/Frogging-Family/nvidia-all/b144bdc05c4d052add4d0e3e7844d2dadf2d43b1/patches/linux-version.diff";
          sha256 = "12ybid6fvmakfkidfp5m7lr5h3wpb0mr944s36y066krvmzi1gir";
        })
        (fetchpatch {
          url = "https://raw.githubusercontent.com/Frogging-Family/nvidia-all/b144bdc05c4d052add4d0e3e7844d2dadf2d43b1/patches/kernel-5.6.patch";
          sha256 = "05dljqgkyb2624461jmn9qpj6jj6diaww6zizzarizhyv78fq8bs";
        })
      ];
```

---

I believe you'd need to `sed s/kernel-dkms/kernel/` these if you're downloading them from the source.

---

@arcnmx is correct

---

Okay, then:
```suggestion
      patches = [
        (fetchpatch {
          url = "https://raw.githubusercontent.com/Frogging-Family/nvidia-all/master/patches/linux-version.diff";
          sha256 = "0xp76ydxjf1ghpnl733gw9lx7428zxp2ys2ssw518xqpkkb8vnqp";
          stripLen = 1;
          extraPrefix = "kernel/";
        })
        (fetchpatch {
          url = "https://raw.githubusercontent.com/Frogging-Family/nvidia-all/master/patches/kernel-5.6.patch";
          sha256 = "0blpgrjdh72k6gzkqs5l0i57bz0z4vi5519bl07l9vx1ws1ssahc";
          stripLen = 1;
          extraPrefix = "kernel/";
        })
      ];
```
Anything else?

---

You want `stripLen=2` (then fix hashes), but otherwise this approach sounds good to me.

---

Done, thanks!